### PR TITLE
Restore list format defaulting

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
Fixes SMW 3.0 regression (by restoring 2.5.x behavior)

Reported in https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/4363

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Tested manually with `{{#ask: [[{{PAGENAME}}]] |?IsNotSet=| mainlabel=- |default=hello }}`. Still needs regression test but I won't have time for that today.

**Comment by JeroenDeDauw** [comment]
As per comments on https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/4363, this PR only fixes plainlist. The other list formats are still broken. That ought to not prevent this partial fix from being merged.

**Comment by kghbln** [comment]
no bloody idea why the tests fail

**Comment by kghbln** [comment]
Ok, I finally managed to get the tests working. As a result we need a squash and merge.

File: src/Query/ResultPrinters/ListResultPrinter.php [hunk 1]
```diff
@@ -55,12 +55,17 @@ public function isDeferrable() {
 	 * @return string
 	 */
 	protected function getResultText( SMWQueryResult $queryResult, $outputMode ) {
-
 		$builder = $this->getBuilder( $queryResult );
 
 		$this->hasTemplates = $this->hasTemplates();
 
-		return $builder->getResultText() . $this->getFurtherResultsText( $queryResult, $outputMode );
+		$result = $builder->getResultText() . $this->getFurtherResultsText( $queryResult, $outputMode );
```
Comment (s7eph4n) [inline src/Query/ResultPrinters/ListResultPrinter.php]: What about just asking for `$queryResult->getCount()` here and not even calling the builder? 
The question is, what is expected for list/ol/ul for getCount === 0. Just an empty string or a wrapped empty string?

Comment (JeroenDeDauw) [reply src/Query/ResultPrinters/ListResultPrinter.php]: `$queryResult->getCount()` is already compared to zero before this method is invoked. That just checks if there are no subjects though. The defaulting should also happen when there are subjects but no values.

If there are no values the default should be shown, at least if we want to restore the old behavior. Seems sensible enough to me.

**Pull request merged**